elasticsearch/templates/_helpers.tpl

```diff
@@ -27,6 +35,14 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- end -}}
 {{- end -}}
 
+{{- define "masterHeadlessService" -}}
+{{- if contains "." (include "masterService" .) -}}
```

Comment:
This only covers use-cases where someone is migrating from a different namespace. In the [migration guide](https://github.com/elastic/helm-charts/tree/master/elasticsearch/examples/migration) the recommendation is to just manually set the environment variable.

https://github.com/elastic/helm-charts/blob/97f0004c50bdda7bee34069d2fe06f4eead96207/elasticsearch/examples/migration/master.yml#L9-L11

I think it would better to use this method as it works for all use cases, and doesn't add any extra complexity into the templating. Especially for something that only needs to exist temporarily and be done once for migrating users.